### PR TITLE
Issue #33 #32 Fix gh-release and added release instructions

### LINKED ISSUE #32: Add release instructions to README

### PR DESCRIPTION
**Pull request by brgnepal**
Fix #32  and #33 

cc @hferentschik @LalatenduMohanty 

Please check in Mac by release to your own fork.

### CODE REVIEW

**Comment by brgnepal** [comment]
I am not able to test in Linux OS too. Getting some `gh-release` specific error locally: 
```
$ make release
go get -u github.com/progrium/gh-release/...
# rm -rf release &&
mkdir -p release
cp /home/budhram/redhat/minishift-b2d-iso/build/minishift-b2d.iso release/
gh-release checksums sha256
/tmp/bashenv.433487048: line 63: syntax error near unexpected token `('
/tmp/bashenv.433487048: line 63: `export BASH_FUNC_module()=$'() {  eval `/usr/bin/modulecmd bash $*`'
/home/budhram/.basher/bash: main: command not found
Makefile:35: recipe for target 'release' failed
make: *** [release] Error 127
```

**Comment by brgnepal** [comment]
Definitely some issue with `gh-release` with its dependency in Windows OS. Probably might be the reason, the author is also not publishing windows artifacts.

```
PS C:\minishift-b2d-iso> make release
go get -u github.com/progrium/gh-release/...
# github.com/progrium/go-basher
C:\gowork\src\github.com\progrium\go-basher\basher.go:52: undefined: RestoreAsset
make: *** [C:\gowork/bin/gh-release] Error 2
```

File: Makefile [hunk 1]
```diff
@@ -23,22 +23,19 @@ clean:
...
-get_gh-release: init
-	curl -sL https://github.com/progrium/gh-release/releases/download/v2.2.1/gh-release_2.2.1_linux_x86_64.tgz > $(BUILD_DIR)/gh-release_2.2.1_linux_x86_64.tgz
-	tar -xvf $(BUILD_DIR)/gh-release_2.2.1_linux_x86_64.tgz -C $(BUILD_DIR)
-	rm -fr $(BUILD_DIR)/gh-release_2.2.1_linux_x86_64.tgz
+$(GOPATH)/bin/gh-release:
+	go get -u github.com/progrium/gh-release/...
 
 .PHONY: release
-release: iso get_gh-release
+release: iso $(GOPATH)/bin/gh-release
```
Comment (hferentschik) [inline Makefile:34]: :+1

File: Makefile [hunk 2]
```diff
@@ -23,22 +23,19 @@ clean:
...
 	rm -rf release && mkdir -p release
 	cp $(BUILD_DIR)/minishift-b2d.iso release/
-	$(BUILD_DIR)/gh-release checksums sha256
-	$(BUILD_DIR)/gh-release create minishift/minishift-b2d-iso $(VERSION) master v$(VERSION)
+	gh-release checksums sha256
```
Comment (hferentschik) [inline Makefile:37]: +1

File: README.md [hunk 3]
```diff
@@ -44,6 +46,28 @@ $ make
...
 - The ISO can be used with [minishift](https://github.com/minishift/minishift) with the help of `--iso-url` i.e. `--iso-url=file:///$PATHTOISO`.
 
+
+<a name="releasing-the-b2d-iso"></a>
+## Releasing the minishift-b2d ISO
+
+<a name="release-prerequisites"></a>
+### Prerequisites
+
+- Installed [Go](https://en.wikipedia.org/wiki/Go_(programming_language)) with `GOPATH` properly set.
```
Comment (hferentschik) [inline README.md]: +1

File: README.md [hunk 4]
```diff
@@ -44,6 +46,28 @@ $ make
...
+- A running docker daemon.
+
+<a name="releasing-the-iso"></a>
+### Releasing the ISO
```
Comment (hferentschik) [inline README.md]: I'll put the releasing part after testing. First you build, then you test and finally you release. I think these are the logical steps.

**Review by hferentschik — COMMENTED** [review]
Looking ok to me. I'd just switch release and test part. 

Also, while on it, do you mind adding the community part as well?

**Comment by hferentschik** [comment]
> Definitely some issue with gh-release with its dependency in Windows OS

I don't think this is a a problem for now. For my part I was able to successfully release to my fork.

File: README.md (hunk 4 above)
Comment (brgnepal) [reply README.md]: Sure :+1:

**Comment by brgnepal** [comment]
> For my part I was able to successfully release to my fork.

Thanks for testing. I am having some issue (https://github.com/minishift/minishift-b2d-iso/pull/53#issuecomment-321790267) locally so couldn't test with my fork but definitely should work as I didn't change anything `gh-release` specific.

File: README.md (hunk 3 above)
Comment (LalatenduMohanty) [reply README.md]: s/Installed/Install/

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

**Comment by hferentschik** [comment]
@LalatenduMohanty, do you mind giving it a try?

File: README.md [hunk 5]
```diff
@@ -64,6 +67,27 @@ If you want to see the logs generated by test, run the test as:
 $ SHOW_LOG=--show-job-log make test
 ```
 
+<a name="releasing-the-b2d-iso"></a>
+## Releasing the minishift-b2d ISO
+
+<a name="release-prerequisites"></a>
+### Prerequisites
+
+- Install [Go](https://en.wikipedia.org/wiki/Go_(programming_language)) with `GOPATH` properly set.
```
Comment (hferentschik) [inline README.md]: "Install" is a verb, you need a noun to fit this enumeration. 

Working installation of ....

File: README.md [hunk 6]
```diff
@@ -80,6 +104,15 @@ On a successful master build, the build artifacts can be found at
 For more information about CentOS CI, check out its [Wiki](https://wiki.centos.org/QaWiki/CI) to
 know more about CentOS CI.
 
+<a name="community"></a>
+## Community
```
Comment (hferentschik) [inline README.md]: I put community last, but I leave it up to you.

**Review by hferentschik — COMMENTED** [review]
I added some more comments. Basically I am ok with this. Just would like to get another opinion from a Linux machine.

**Review by hferentschik — APPROVED** [review]

**Comment by LalatenduMohanty** [comment]
@budhrg I have done previous releases on a CentOS VM and it worked fine. But on Fedora this issue exists https://github.com/progrium/gh-release/issues/19

**Review by LalatenduMohanty — APPROVED** [review]
Works for me on CentOS7 https://github.com/LalatenduMohanty/minishift-b2d-iso/releases/tag/v1.1.0

**Comment by brgnepal** [comment]
With my CentOS VM, I am getting exit code 2 .
```
$ make release
make release
rm -rf release && mkdir -p release
cp /home/budhram/redhat/minishift-b2d-iso/build/minishift-b2d.iso release/
gh-release checksums sha256
Writing sha256 checksum files...
gh-release create budhrg/minishift-b2d-iso 1.1.0 master v1.1.0
Creating release v1.1.0 from branch master ...
make: *** [release] Error 2

```

**Pull request merged**